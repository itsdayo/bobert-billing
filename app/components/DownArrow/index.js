/**
 *
 * DownArrow
 *
 */

import React from 'react';
// import PropTypes from 'prop-types';
import styled from 'styled-components';

function DownArrow(props) {
  return <Arrow {...props} />;
}

const Arrow = styled.i`
  border: solid ${props => (props.secondary ? '#fff' : '#3755b0')};
  border-width: 0 2px 2px 0;
  display: inline-block;
  padding: 3px;
  position: relative;
  bottom: 1px;
  cursor: pointer;
  transform: scaleX(-1);
  -webkit-transform: rotate(45deg);
`;

DownArrow.propTypes = {};

export default DownArrow;
