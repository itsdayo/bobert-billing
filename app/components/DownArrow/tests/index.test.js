/**
 *
 * Tests for DownArrow
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import DownArrow from '../index';

describe('<DownArrow />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<DownArrow />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<DownArrow />);
    expect(firstChild).toMatchSnapshot();
  });
});
