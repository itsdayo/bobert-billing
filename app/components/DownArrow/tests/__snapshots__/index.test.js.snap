// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DownArrow /> Should render and match the snapshot 1`] = `
<i
  class="DownArrow__Arrow-sc-1k5ecwc-0 cJGZyn"
/>
`;
